contracts/contracts/strategies/NativeStaking/ValidatorRegistrator.sol

```diff
@@ -0,0 +1,349 @@
+    // needs to a approve further staking by resetting the stake ETH tally
+    function setStakeETHThreshold(uint256 _amount) external onlyGovernor {
+        stakeETHThreshold = _amount;
+        emit StakeETHThresholdChanged(_amount);
+    }
+
+    /// @notice Reset the stakeETHTally
+    function resetStakeETHTally() external onlyStakingMonitor {
+        stakeETHTally = 0;
+        emit StakeETHTallyReset();
+    }
+
+    /// @notice Stakes WETH to the node validators
+    /// @param validators A list of validator data needed to stake.
+    /// The `ValidatorStakeData` struct contains the pubkey, signature and depositDataRoot.
+    /// Only the registrator can call this function.
+    // slither-disable-start reentrancy-eth
+    function stakeEth(ValidatorStakeData[] calldata validators)
+        external
+        onlyRegistrator
+        whenNotPaused
+    {
+        uint256 requiredETH = validators.length * 32 ether;
+
+        // Check there is enough WETH from the deposits sitting in this strategy contract
+        require(
+            requiredETH <= IWETH9(WETH).balanceOf(address(this)),
+            "Insufficient WETH"
+        );
+        require(
+            activeDepositedValidators + validators.length <= MAX_VALIDATORS,
+            "Max validators reached"
+        );
+
+        require(
+            stakeETHTally + requiredETH <= stakeETHThreshold,
+            "Staking ETH over threshold"
+        );
+        stakeETHTally += requiredETH;
+
+        // Convert required ETH from WETH
+        IWETH9(WETH).withdraw(requiredETH);
+        _wethWithdrawnAndStaked(requiredETH);
+
+        /* 0x01 to indicate that withdrawal credentials will contain an EOA address that the sweeping function
+         * can sweep funds to.
+         * bytes11(0) to fill up the required zeros
+         * remaining bytes20 are for the address
+         */
+        bytes memory withdrawalCredentials = abi.encodePacked(
+            bytes1(0x01),
+            bytes11(0),
+            address(this)
+        );
+
+        // For each validator
+        for (uint256 i = 0; i < validators.length; ++i) {
+            bytes32 pubKeyHash = keccak256(validators[i].pubkey);
+            VALIDATOR_STATE currentState = validatorsStates[pubKeyHash];
+
+            require(
+                currentState == VALIDATOR_STATE.REGISTERED,
+                "Validator not registered"
+            );
+
+            IDepositContract(BEACON_CHAIN_DEPOSIT_CONTRACT).deposit{
+                value: 32 ether
+            }(
+                validators[i].pubkey,
+                withdrawalCredentials,
+                validators[i].signature,
+                validators[i].depositDataRoot
+            );
+
+            validatorsStates[pubKeyHash] = VALIDATOR_STATE.STAKED;
+
+            emit ETHStaked(
+                pubKeyHash,
+                validators[i].pubkey,
+                32 ether,
+                withdrawalCredentials
```

Comment:
These `withdrawalCredentials` are essentially hardcoded and will never change without some fairly major changes to the ethereum. Is there any reason for emitting the withdrawalCredentials?

Reply:
Good point. It's just the strategy proxy address left padded. I'll take it out as it saves 1k in gas